runtime/common/tests/xcm_tests.rs

```diff
@@ -2134,3 +1816,318 @@ fn test_automatic_versioning_on_runtime_upgrade_with_para_b() {
+			AccountId32 {
+				network: NetworkId::Any,
+				id: ALICE.into(),
+			},
+		),
+	};
+
+	// Sending xcm to parachain 5 should not work
+	ParaA::execute_with(|| {
+		assert_noop!(
+			parachain::XTokens::transfer(
+				parachain::Origin::signed(ALICE.into()),
+				parachain::CurrencyId::MantaCurrency(a_currency_id),
+				100,
+				Box::new(VersionedMultiLocation::V1(dest)),
+				80
+			),
+			orml_xtokens::Error::<parachain::Runtime>::NotSupportedMultiLocation,
+		);
+	});
+
+	let x3_dest = MultiLocation {
+		parents: 1,
+		interior: X3(
+			Parachain(2),
+			PalletInstance(PALLET_ASSET_INDEX),
+			AccountId32 {
+				network: NetworkId::Any,
+				id: ALICE.into(),
+			},
+		),
+	};
+	// We don't support X3 or more longer Junctions.
+	ParaA::execute_with(|| {
+		assert_noop!(
+			parachain::XTokens::transfer(
+				parachain::Origin::signed(ALICE.into()),
+				parachain::CurrencyId::MantaCurrency(a_currency_id),
+				100,
+				Box::new(VersionedMultiLocation::V1(x3_dest)),
+				80
+			),
+			orml_xtokens::Error::<parachain::Runtime>::NotSupportedMultiLocation,
+		);
+	});
+
+	let parents_as_2_relay_dest = MultiLocation {
+		parents: 2,
+		interior: X1(AccountId32 {
+			network: NetworkId::Any,
+			id: ALICE.into(),
+		}),
+	};
+	// relaychain dest with wrong parents should not work.
+	ParaA::execute_with(|| {
+		assert_noop!(
+			parachain::XTokens::transfer(
+				parachain::Origin::signed(ALICE.into()),
+				parachain::CurrencyId::MantaCurrency(a_currency_id),
+				100,
+				Box::new(VersionedMultiLocation::V1(parents_as_2_relay_dest)),
+				80
+			),
+			orml_xtokens::Error::<parachain::Runtime>::NotSupportedMultiLocation,
+		);
+	});
+
+	let parents_as_2_dest = MultiLocation {
+		parents: 2,
+		interior: X2(
+			Parachain(2),
+			AccountId32 {
+				network: NetworkId::Any,
+				id: ALICE.into(),
+			},
+		),
+	};
+	// Wrong parents should not work.
+	ParaA::execute_with(|| {
+		assert_noop!(
+			parachain::XTokens::transfer(
+				parachain::Origin::signed(ALICE.into()),
+				parachain::CurrencyId::MantaCurrency(a_currency_id),
+				100,
+				Box::new(VersionedMultiLocation::V1(parents_as_2_dest)),
+				80
+			),
+			orml_xtokens::Error::<parachain::Runtime>::NotSupportedMultiLocation,
+		);
+	});
+
+	let here_dest = MultiLocation {
+		parents: 1,
+		interior: Here,
+	};
+	// The destination with no receiver should not work.
+	ParaA::execute_with(|| {
+		assert_noop!(
+			parachain::XTokens::transfer(
+				parachain::Origin::signed(ALICE.into()),
+				parachain::CurrencyId::MantaCurrency(a_currency_id),
+				100,
+				Box::new(VersionedMultiLocation::V1(here_dest)),
+				80
+			),
+			orml_xtokens::Error::<parachain::Runtime>::NotSupportedMultiLocation,
+		);
+	});
+}
+
+#[test]
+fn less_than_min_xcm_fee_should_not_work() {
+	MockNet::reset();
+
+	let para_a_id = 1;
+	let para_b_id = 2;
+	let para_a_source_location = create_asset_location(1, para_a_id);
+	let para_b_source_location = create_asset_location(1, para_b_id);
```

Comment:
`para_b_source_location` should be some X2, or X3 location type. so the difference between the asset location and the reserve location (that you write to MinXcmFeeOfReserves) is better illustrated.